src/dsp_tools/utils/id_to_iri.py

```diff
@@ -207,5 +239,10 @@ def id_to_iri(
         tree=tree,
         mapping=mapping,
     )
+    if remove_resource_if_id_in_mapping:
+        tree, success = _remove_resources_if_id_in_mapping(
+            tree=tree,
+            mapping=mapping,
+        )
     _write_output_file(orig_xml_file=xml_file_as_path, tree=tree)
     return success
```

Comment:
Is it on purpose that success can be overwritten by the method '_remove_resources_if_id_in_mapping' which follows?